Makefile

```diff
@@ -284,7 +284,7 @@ ifdef LUA
     # Windows expects to have lua unpacked at a specific location
     LDFLAGS += -llua
   else
-    LUA_CANDIDATES = lua5.2 lua-5.2 lua5.1 lua-5.1 lua
+    LUA_CANDIDATES = $(basename $(LUA_BINARY)) lua5.2 lua-5.2 lua5.1 lua-5.1 lua
```

Comment:
I said "It is a real problem", you just cannot build against luajit atm without modifying the makefile.

That said this entire system has some problems, if you'd like we can postpone this commit and I'll work up something more robust.